arkane/encorr/reference.py

```diff
@@ -665,5 +669,57 @@ def _is_valid_calculated_data(data_dictionary):
     return False
 
 
+def update_reference_db(path: str, model_chemistry: str = None, ref_paths: List[str] = None) -> None:
+    """
+    Update the reference species database from a collection of ArkaneSpecies YAML files. All of the YAML files should
+    be included in a single folder.
+
+    Notes:
+        ArkaneSpecies YAML files must end in either '.yml' or '.yaml'
+
+    Args:
+        path (str): The path to a directory containing all of the ArkaneSpecies YAML files
+        model_chemistry (str): Model chemistry string matching the Arkane format to be updated. If not supplied will
+            default to the ``level_of_theory`` parameter of the ArkaneSpecies objects
+        ref_paths (list): A list to the paths of reference sets to load. By default only the main reference set is
+            loaded
+
+    Returns:
+        None
+    """
+    spcs_list = []
+    for f in os.listdir(path):
+        if any(file_extension in f.lower() for file_extension in ['.yml', '.yaml']):
+            spc = ArkaneSpecies.__new__(ArkaneSpecies)
+            spc.load_yaml(path=os.path.join(path, f))
+            spcs_list.append(spc)
+
+    database = ReferenceDatabase()
+    database.load(ref_paths)
+    ref_spcs_list = [ref_spcs for ref_set in database.reference_sets.values() for ref_spcs in ref_set]
+
+    for arkane_spc in spcs_list:
+        species = Species().from_adjacency_list(arkane_spc.adjacency_list, raise_atomtype_exception=False,
+                                                raise_charge_exception=False)
+        # Try to generate resonance structures
+        try:
+            species.generate_resonance_structures()
+        except (AtomTypeError, ValueError):  # This can fail for charged or unusual species. Move on without it
+            pass
+
+        for ref_spcs in ref_spcs_list:
+            if ref_spcs.is_isomorphic(species, generate_resonance_structures=False):
```

Comment:
I'm not confident that `is_isomorphic` will work. From what I can tell, `is_isomorphic` does not consider charge, so would, for example, methylium ([CH3+]) and methanide ([CH3-]) be isomorphic?

Reply:
Interesting, I forgot to check to see if charge was considered in the isomorphism checks. When I was loading in Duminda's calc last night though, I did check that all of the species got mapped to one and only one reference species, so it seems like it was able to differentiate somehow. I'll investigate further, but at the very least add a check for charge (and multiplicity if needed while I am at it) to the isomorphism check method